cpp/ps0008/p0008.cpp

```diff
@@ -1,31 +1,46 @@
 #include <iostream>
-#include <limits>
 #include <cctype>
 #include <string>
 using namespace std;
+
+// Decrypting process 3/3
 void decrypt(string msg, int key)
 {
     for (char& c : msg) {
-        if (isupper(c)) {    
+        if (isupper(c)) {
             c = (c - 'A' - key + 26) % 26 + 'A';
         } else if (islower(c)) {
             c = (c - 'a' - key + 26) % 26 + 'a';
         }
     }
     cout << "Decrypted message: " << msg << endl;
 }
+
 int main()
 {
+    // prompting user for valid ciphertext and key
     string ciphertext;  
-    int key;            
+    float key; // to validate floating-point numbers for key
+    
+    // Taking message 1/1
     cout << "Enter the ciphertext: ";
-    getline(cin, ciphertext);//عشان يقرا السطر كامل 
-    cout << "Enter the key: ";
-    while (!(cin >> key)) {
-        cout << "Invalid input. Please enter an integer for the key: ";
-        cin.clear();  //عشان امسح القيم الغير صحيحه
-       cin.ignore(100, '\n'); 
+    getline(cin, ciphertext); //عشان يقرا السطر كامل 

```

Comment:
- [x] prompted user for ciphertext (encrypted message)
- [x] stored the targeted ciphertext inside a string variable